gensim/models/coherencemodel.py

```diff
@@ -441,12 +441,16 @@ def topics(self, topics):
         self._topics = new_topics
 
     def _ensure_elements_are_ids(self, topic):
-        try:
-            return np.array([self.dictionary.token2id[token] for token in topic])
-        except KeyError:  # might be a list of token ids already, but let's verify all in dict
-            topic = (self.dictionary.id2token[_id] for _id in topic)
-            return np.array([self.dictionary.token2id[token] for token in topic])
-
+        tokens = [t for t in topic if t in self.dictionary.token2id]
```

Comment:
I think the naming here is very confusing, and it's possible to do better.

```python
def _ensure_elements_are_ids(self, topic):
  topic_is_tokens = [self.dictionary.token2id[t] for t in topic if t in self.dictionary.token2id]
  topic_is_ids = [i for i in topic if i in self.dictionary.id2token]
  if len(topic_is_tokens) > len(topic_is_ids):
    return np.array(topic_is_tokens)
  elif len(topic_is_ids) > len(topic_is_tokens):
    return np.array(topic_is_ids)
  else:
    raise ValueError('unable to interpret token as either a list of tokens or a list of ids')
```

Reply:
Solved also this